src/modules/signer/signer.service.ts

```diff
@@ -53,15 +54,18 @@ export class SignerService {
         this._servicesInitializers.push(initializer);
     }
 
-    on(event: ProviderEvent, cb) {
-        const provider = this._signer.provider;
-        if (provider instanceof EventEmitter) {
-            provider.on(event, cb);
-        } else if (provider instanceof WalletConnectProvider) {
-            provider.on(event, cb);
+    async emit(e: ProviderEvent) {
```

Comment:
I'm a bit confused by the naming here. It seems like this more of a `handleEvent` function 🤔 ? It doesn't look like anything is actually being emitted?

Reply:
It is emitted in a sense that it notifies registered listeners. It is similar to Node.js EventEmiter.emit